### DISCUSSION TITLE
Opting-in to, emulating or supporting pre-12 style formatting

### DISCUSSION THREAD

**Discussion by riotrah** [sql-formatter-org/sql-formatter#587] [Q&A] [answered]
Hello! Thanks so much for your work on this.

I want to make the business case for upgrading our version of this lib (from v2 no less) especially for `formatDialect` etc, and it would be all the better if it was possible to at least _demonstrate_ parity with the older multiline-inclined formatting style.

Is this possible? I'm imagining we'd likely prefer the newer style anyways, but it minimizes the risk factor of upgrading, for whatever it's worth.

**Comment by nene** [comment, marked as answer]
That's not really feasible. It would be a major development effort to bring this back in some configurable form. Plus the change is pretty fuzzy. It affects some SQL statements a lot while not effecting others not at all.

For what it's worth, I can say that nobody has so far complained about the formatting style change.

If you want to be extra safe you can just upgrade to version 11. Then again... there have been many breaking changes since 2.x.

**Reply by riotrah** [reply]
Sure, fair enough! Thanks either way, this and your CST project are very cool,
